### ISSUE TITLE
Libsass support?

### ISSUE THREAD

**Issue by leetomblin** [eduardoboucas/include-media#10]
Currently getting this when trying to compile:

```
stylesheets/vendor/include-media:17: error: error reading values after phone
```

I realise this is because Libsass doesn't support list maps - any work around for this? I've got quite used to my Sass compiling near enough instantly :)

**Comment by eduardoboucas** [comment]
What version of Libsass are you using?

**Comment by klapec** [comment]
Seems like it works fine with libsass. I just tested it and haven't got any errors. And according to this: http://sass-compatibility.github.io/#map_data_type libsass does support Sass Map API.

**Comment by eduardoboucas** [comment]
Exactly. It's supposed to work with libsass >= 3.1.

@leetomblin If you're using an older version of libsass and you really want to stick with it, maybe you can try [this polyfill](https://github.com/lunelson/sass-list-maps).

**Comment by leetomblin** [comment]
Sorry - should have mentioned I'm using the grunt version of Libsass https://github.com/sindresorhus/grunt-sass

I tried the polyfills, but ended up with even more errors - although I was probably doing something wrong. Will give it another try, thanks.

**Comment by danny-englander** [comment]
I'm using LibSass as well (grunt-sass) and just ran into this error:

```
Warning: include-media:97: only variable declarations and control directives are allowed inside functions
Use --force to continue.
```
I am using the latest version of grunt-sass so I don't know if I should open a new issue. I've been able to stop using the polyfill for maps with the latest updates to LibSass as well.

**Comment by eduardoboucas** [comment]
@danny-englander Thanks for raising the problem. I've reopened the issue and I'll look into this.

**Comment by eduardoboucas** [comment]
@danny-englander You're using `grunt-sass` and not `grunt-contrib-sass`, right? The latter works fine for me.

**Comment by danny-englander** [comment]
@eduardoboucas yes grunt-sass which is a LIbSass version for node-sass (LibSass)

**Comment by eduardoboucas** [comment]
It all comes back to support for maps, which still seems to be problematic in libsass. 

@klapec you mentioned that you were getting no errors using libsass. Which version are you using? Are you using any wrapper?

Maybe the problem is with [node-sass](https://github.com/sass/node-sass), the wrapper that [grunt-sass](https://github.com/sindresorhus/grunt-sass) is using? Could it be using a different/older version of libsass?

We can try to find exactly what versions are problematic, but in the end I'm afraid there's not much I can do since it's such a structural thing. If maps aren't properly supported, this library will never work.

**Comment by danny-englander** [comment]
@eduardoboucas Hey thanks for looking in to it! I'll circle back and see if the polyfill will work.  In the mean time I am using the new ```susy-media()``` mixin that Susy for LIbSass has and it works well. This was more of a thing that I wanted to test out but I am pretty much all LibSass going forward just because of the instant compile times.

**Comment by klapec** [comment]
> @klapec you mentioned that you were getting no errors using libsass. Which version are you using? Are you using any wrapper?

I was using gulp-sass at the time. I'll test it out tomorrow morning.

**Comment by klapec** [comment]
Looks like something was recently changed in include-media that doesn't quite work with libsass. I had an older version of the mixin (https://gist.github.com/klapec/83b1b471bd9c5f935203 - not sure which version it is) that DOES work with libsass, but the newest one doesn't.

```
(...) assets/src/stylesheets/vendors/include-media:97: only variable declarations and control directives are allowed inside functions
```

**Comment by eduardoboucas** [comment]
@klapec That's a huge help, I will look into this.

Thanks guys!

**Comment by eduardoboucas** [comment]
This seems to be a problem with grunt-sass, not libsass itself. The errors you're seeing now, @klapec and @danny-englander, happen on every instance of `@error`. We've added some error handling on the latest version and that seems to be causing you problems. If you try commenting those out (obviously not a solution), it should compile fine.

However, include-media was working before for @klapec but not for @leetomblin, and my guess is that @klapec only tested the mixin with set breakpoints (e.g. `@include media(">phone", "<tablet") {}`) whereas @leetomblin was throwing custom values to the mix (e.g. `@include media(">phone", "<850px") {}`) and this was actually causing a different problem.

Because grunt-sass is using a beta and unstable version of libsass, the function that parses custom values doesn't seem to work. I'm not sure there's anything we can do to overcome this, as again we're dealing with a beta version of libsasss used by grunt-sass. I'll keep digging into this though.

Thanks guys for your feedback and thanks @HugoGiraudel for helping me figure this out.

**Comment by eduardoboucas** [comment]
Okay, I know what's causing this and I'll commit a fix later today.

Thanks @HugoGiraudel for your precious help!

**Comment by eduardoboucas** [comment]
I've pushed a new branch (grunt-sass-fix) which will hopefully fix the issues.

@klapec @danny-englander @leetomblin could you possibly try using [this file](https://raw.githubusercontent.com/eduardoboucas/include-media/grunt-sass-fix/dist/_include-media.scss) and let me know if you still see any errors? If not, I'll merge this into master.

Thanks!

**Comment by danny-englander** [comment]
@eduardoboucas - I just tested this out with grunt-sass and it works great. I even tried my own defined breakpoint variable:

 ```
@include media(">=tablet", "<#{$bp-large}") {
    // Your rules here
}
```

Thank you!

**Comment by eduardoboucas** [comment]
Great! Let's wait for feedback from the other folks and if everyone is happy I'll merge this in.

**Comment by leetomblin** [comment]
Working great here too.

Thanks for this!

**Comment by eduardoboucas** [comment]
Fixed in [#20](https://github.com/eduardoboucas/include-media/pull/20)
